### PR TITLE
Backend Auth, Frontend Login / Account Creation form

### PR DESCRIPTION
**Pull request by T7Q**
hi! few new features from my side :)

backend:
- /auth/login route
- added authMiddleware. authRequired can be added to all routes that need authentication check, this middleware saves logged user id to "req.user". 
- extended Diana's middleware to validate login form input, also fixed some small bugs in validateAccountCreation, @DianaMukaliyeva take a look

frontend: 
- added Login and Create User Forms and connected them with backend 
- updated routing and moved videoPlayer component to modal window. @iljaSL take a look, you can view your component by clicking "MOVIE HERE" button

### CODE REVIEW

File: backend/utilities/authMiddleware.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import jwt from 'jsonwebtoken';
```
Comment (DianaMukaliyeva) [inline backend/utilities/authMiddleware.js:1]: what do you think if we place this functions to middleware.js file?

File: frontend/public/locales/ru/translation.json [hunk 2]
```diff
@@ -19,6 +19,7 @@
         "minLen": "минимальное кол-во знаков {{number}}",
         "maxLen": "must be max {{number}} characters",
         "nameFormat": "must contain only letters and numbers",
+        "passwordFormat": "must contain at least 1 upper case, 1 lower case, 1 number",
```
Comment (lapaset) [inline frontend/public/locales/ru/translation.json]: Should this be in russian?

File: frontend/src/App.js [hunk 3]
```diff
@@ -1,52 +1,45 @@
...
     setAuthToken(localStorage.getItem('token'));
-    // TO DO finish when backend is done
-    // try {
-    //   const res = await authService.auth();
-    //   setUser({userId: res.userId, lang: res.lang})
-    // } catch (exception) {
-    // }
+    if (localStorage.getItem('token')) {
+      const decoded = jwt_decode(localStorage.getItem('token'));
+      setUser({ userId: decoded.id, lang: decoded.lang });
```
Comment (lapaset) [inline frontend/src/App.js:24]: Do you need to check` if (decoded && decoded.id && decoded.lang)`?

File: frontend/public/locales/ru/translation.json (hunk 2 above)
Comment (T7Q) [reply frontend/public/locales/ru/translation.json]: good point, will update it

File: backend/utilities/authMiddleware.js (hunk 1 above)
Comment (T7Q) [reply backend/utilities/authMiddleware.js:1]: works for me

**Review by lapaset — APPROVED** [review]
Looks good to me!

**Pull request merged**